src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1067,7 +1073,7 @@ private ImmutableArray<BoundNode> GuardedAddBoundTreeAndGetBoundNodeFromMap(CSha
 
             if (bound != null)
             {
-                alreadyInTree = _guardedNodeMap.ContainsKey(bound.Syntax);
+                alreadyInTree = _guardedNodeMap.ContainsKey(AsNodeMapKey(bound.Syntax));
```

Comment:
I agree with @AlekseyTs. In general, it is going to be difficult to ensure consistency between bound tree structures in VB and C#, it could be in place for performance reasons. Additionally, we could also break lowering passes in doing so. The bound node tree structure can also undergo any changes in future - its not a public API, but an implementation detail. The right approach would be to handle these differences at the layer that maps syntax/bound nodes to IOperations. We should write more unit tests to uncover these differences, and prevent any public facing regressions.